fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -96,18 +97,20 @@ public boolean doesResourceExist(final Transaction transaction, final FedoraId f
             // Root always exists.
             return true;
         }
-        if (!fedoraId.isMemento()) {
-            // containment index doesn't handle versions, so don't bother checking for them.
+        if (!(fedoraId.isMemento() || fedoraId.isAcl())) {
+            // containment index doesn't handle versions and only tells us if the resource (not acl) is there,
+            // so don't bother checking for them.
             final String transactionId = transaction == null ? null : transaction.getId();
             return containmentIndex.resourceExists(transactionId, fedoraId);
         } else {
 
             final PersistentStorageSession psSession = getSession(transaction);
 
             try {
-                final String id = fedoraId.isDescription() ? fedoraId.getDescriptionId() : fedoraId.getResourceId();
-                psSession.getHeaders(id, fedoraId.getMementoInstant());
-                return true;
+                // Resource ID for metadata or ACL contains their individual endopoints (ie. fcr:metadata, fcr:acl)
+                final String id = fedoraId.getResourceId();
+                final ResourceHeaders headers = psSession.getHeaders(id, fedoraId.getMementoInstant());
```

Comment:
In the case that the resource is an ACL (see line:100 above), there may not be a "memento instant", no?

Reply:
Yes and the `getMementoInstant()` function would return `null` in that case.